### ISSUE TITLE
Cannot restrict pad types to registered users only

### ISSUE THREAD

**Issue by henhuy** [xwiki-labs/cryptpad#303]
In my `customize/application_config.js`, I have follwing config:
```
config.availablePadTypes = ['drive', 'pad', 'code', 'slide', 'poll', 'whiteboard', 'file', 'todo', 'contacts'];
config.registeredOnlyTypes = ['drive', 'pad', 'code', 'slide', 'poll', 'whiteboard', 'file', 'todo', 'contacts'];
```
in order to avoid guest pads. But still all pads are listed and available from home page or by adding pad type to url (i.e. `.../pad`, or `.../poll`) as anonymous user. 
Also, I noticed when using default config:
```
config.registeredOnlyTypes = ['file', 'contacts'];
```
trying to open a `file pad` gives _correct_ warning: "You must be logged in to upload files", but I can still open `contacts pad` as anonymous without any error warning.

Am I missing something?

**Comment by ansuz** [comment]
Hi,
I agree that the names of the values are misleading. The comment in `application_config_internal.js` indicates what `availablePadTypes` is used for, namely: 

> Select the buttons displayed on the main page to create new collaborative sessions

I don't recall what `registeredOnlyTypes` was intended to do, since you're right that only those two applications do any check to confirm whether they should be accessible or not.

Unfortunately, at the moment our roadmap is very full, and our team is planning to take holidays soon. This functionality is likely to be implemented in a future release, but I can't offer a definite timeline.

**Comment by ansuz** [comment]
As it turns out this wasn't that difficult to implement. We have a fix in our staging branch, which we'll merge to master and deploy on our next release, which is planned for January 15th.

**Comment by henhuy** [comment]
oh nice! thank you.

**Comment by sairim** [comment]
As an aside, while you wait henhuy you can achieve similar functionality by adding something like this to your `application_config.js`:
```
AppConfig.beforeLogin = function(isLoggedIn, callback) {
        if (!isLoggedIn) {
            window.location = 'https://cryptpad.example.net/login';
        }
        callback();
    }; 
    return AppConfig;
```
However, since this is just JS, keep in mind that someone determined enough could get around this quite easily.

**Comment by henhuy** [comment]
Works perfectly! I will use this as my workaround for the moment. Thanks @sairim

**Comment by ansuz** [comment]
This was included in [v2.15.0](https://github.com/xwiki-labs/cryptpad/releases/tag/2.15.0).
